### PR TITLE
add support for Go 1.20

### LINKED ISSUE #3630: Support go1.20
Go 1.20 is expected to be [released in February 2023](https://tip.golang.org/doc/go1.20), but I did not see any issue tracker for supporting it, thus creating one. Thanks!

### PR DESCRIPTION
**Pull request by marten-seemann**
Fixes #3630.

### CODE REVIEW

**Review by lucas-clemente — APPROVED** [review]

File: .github/workflows/cross-compile.yml
Comment (mosajjal) [inline .github/workflows/cross-compile.yml:7]: hate to be that guy but 1.20.0-rc.2 got released today :)

Comment (marten-seemann) [reply .github/workflows/cross-compile.yml:7]: Yes, I'll update this before the PR gets merged.

This PR is currently (soft-) blocked on https://github.com/lucas-clemente/quic-go/pull/3636 though.

Comment (marten-seemann) [reply .github/workflows/cross-compile.yml]: Updated the PR. Fortunately, there were no changes to crypto/tls since rc1, so we don't have to update qtls.

**Pull request merged**
